impl/game_client_lib/client_network_connection.cpp

```diff
@@ -30,14 +28,21 @@ ClientNetworkConnection::ClientNetworkConnection(std::string const& ip, std::uin
 
 ClientNetworkConnection::~ClientNetworkConnection()
 {
-    m_logger.debug(
-        "ClientNetworkConnection destructor called", { "network", "ClientNetworkConnection" });
-    m_logger.verbose("close socket", { "network", "ClientNetworkConnection" });
-    m_socket->close();
-    m_logger.verbose("stop thread", { "stop thread", "ClientNetworkConnection" });
-    stopThread();
-    m_logger.verbose("reset socket pointer", { "network", "ClientNetworkConnection" });
-    m_socket.reset();
+    try {
+        m_logger.debug(
+            "ClientNetworkConnection destructor called", { "network", "ClientNetworkConnection" });
+
+        m_logger.debug("stop thread", { "stop thread", "ClientNetworkConnection" });
+        stopThread();
```

Comment:
Order in this function was wrong. `stopThread()` needs to be called before closing the socket.